From: svc_sdk <svc_sdk@arm.com>
Date: Thu Jul 02 00:00:00 2026
Subject: llvm-changes-for-model-converter-02-07-2026

diff --git a/mlir/include/mlir/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosa.h b/mlir/include/mlir/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosa.h
index 9a42cf75f187..ed56a4f58bb6 100644
--- a/mlir/include/mlir/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosa.h
+++ b/mlir/include/mlir/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosa.h
@@ -42,7 +42,7 @@ constexpr llvm::StringLiteral graphARMGraphConstantIdAttrName =
     "grapharm.graph_constant_id";
 
 std::unique_ptr<Pass> createTosaToSPIRVTosaMarkGraphConstants();
-std::unique_ptr<Pass> createTosaToSPIRVTosa();
+std::unique_ptr<Pass> createTosaToSPIRVTosa(bool analysis = false);
 
 spirv::VerCapExtAttr getDefaultVerCapExtAttr(MLIRContext *context);
 
diff --git a/mlir/lib/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosaPass.cpp b/mlir/lib/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosaPass.cpp
index 815f2db74ffa..b38c0b35e912 100644
--- a/mlir/lib/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosaPass.cpp
+++ b/mlir/lib/Conversion/TosaToSPIRVTosa/TosaToSPIRVTosaPass.cpp
@@ -162,6 +162,8 @@ LogicalResult verifyGraphConstantIdAttrs(Operation *op) {
 }
 
 struct TosaToSPIRVTosa final : impl::TosaToSPIRVTosaBase<TosaToSPIRVTosa> {
+  TosaToSPIRVTosa(bool analysis = false) : analysis(analysis) {}
+
   void runOnOperation() override {
     MLIRContext *context = &getContext();
     RewritePatternSet patterns(context);
@@ -212,12 +214,31 @@ struct TosaToSPIRVTosa final : impl::TosaToSPIRVTosaBase<TosaToSPIRVTosa> {
 
     FrozenRewritePatternSet frozenPatterns(std::move(patterns));
 
+    if (analysis) {
+      DenseSet<Operation *> convertedOps;
+      ConversionConfig config;
+      config.legalizableOps = &convertedOps;
+      if (failed(
+              applyAnalysisConversion(op, *target, frozenPatterns, config))) {
+        signalPassFailure();
+        return;
+      }
+
+      for (Operation *convertedOp : convertedOps) {
+        if (isa<TosaDialect>(convertedOp->getDialect()))
+          llvm::errs() << "Successfully lowered: " << convertedOp->getName()
+                       << " at " << convertedOp->getLoc() << "\n";
+      }
+    }
+
     if (failed(applyPartialConversion(op, *target, frozenPatterns))) {
       signalPassFailure();
     }
   }
 
 private:
+  bool analysis;
+
   IntegerType convertIntegerType(IntegerType integerType) {
     if (integerType.getWidth() == 48) {
       return IntegerType::get(&getContext(), 64, integerType.getSignedness());
@@ -276,8 +297,8 @@ private:
 };
 } // namespace
 
-std::unique_ptr<Pass> createTosaToSPIRVTosa() {
-  return std::make_unique<TosaToSPIRVTosa>();
+std::unique_ptr<Pass> createTosaToSPIRVTosa(bool analysis) {
+  return std::make_unique<TosaToSPIRVTosa>(analysis);
 }
 
 } // namespace tosa
